hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -1055,17 +914,15 @@ private Map<String, HoodieRecord<HoodieMetadataPayload>> fetchBaseFileAllRecords
       return Collections.emptyMap();
     }
 
-    ClosableIterator<HoodieRecord<?>> records = reader.getRecordIterator();
+    ClosableIterator<HoodieRecord<?>> recordIterator = reader.getRecordIterator();
```

Comment:
can we rename fetchBaseFileAllRecordsByPayload to `fetchBaseFileAllRecordsByPayloadForSecIndex` 
to call out that this is some special handling just for SI. 
I do not want someone else to re-use this method for something else down the line